### PR TITLE
feat: add scoringMode to AuditResult

### PR DESCRIPTION
**Pull request by patrickhulce**

### CODE REVIEW

File: lighthouse-cli/types/types.ts [hunk 1]
```diff
@@ -19,6 +19,7 @@ interface AuditFullResult {
   displayValue: string;
   debugString?: string;
   score: boolean|number;
```
Comment (brendankenny) [inline lighthouse-cli/types/types.ts]: Should add it to the closure typedef too

**Comment by brendankenny** [comment]
how do you feel about `scoreType`?

File: lighthouse-core/audits/byte-efficiency/total-byte-weight.js [hunk 2]
```diff
@@ -45,6 +45,7 @@ class TotalByteWeight extends Audit {
           'Network transfer size [costs users real dollars](https://whatdoesmysitecost.com/) ' +
           'and is [highly correlated](http://httparchive.org/interesting.php#onLoad) with long load times. ' +
           'Try to find ways to reduce the size of required files.',
+      scoringMode: 'numeric',
```
Comment (paulirish) [inline lighthouse-core/audits/byte-efficiency/total-byte-weight.js]: should we use an enum?   `scoringMode: Audit.ScoringModes.Numeric`

Comment (patrickhulce) [inline lighthouse-core/audits/byte-efficiency/total-byte-weight.js]: done

**Comment by ebidel** [comment]
Double LGTM

File: lighthouse-core/audits/audit.js [hunk 3]
```diff
@@ -27,6 +27,16 @@ class Audit {
   }
 
   /**
+   * @return {{Numeric: string, Binary: string}}
+   */
+  static get ScoringModes() {
+    return {
+      Numeric: 'numeric',
```
Comment (brendankenny) [inline lighthouse-core/audits/audit.js]: should be `NUMERIC` and `BINARY` or `numeric` and `binary` IMO. also `SCORING_MODES`?

Comment (patrickhulce) [inline lighthouse-core/audits/audit.js]: done

**Comment by ebidel** [comment]
Triple LGTM

**Comment by paulirish** [comment]
lol u guys and the yelling.  lgtm

**Pull request merged**